frontend/src/metabase/public/LocaleProvider.tsx

```diff
@@ -15,7 +15,10 @@ export const LocaleProvider = ({
     if (locale) {
       setIsLoadingLocale(true);
       setLocaleHeader(locale);
-      loadLocalization(locale).then(() => setIsLoadingLocale(false));
+      loadLocalization(locale).then(translatedObject => {
+        setIsLoadingLocale(false);
+        window.MetabaseUserLocalization = translatedObject;
```

Comment:
![Screenshot 2024-11-04 at 10 46 55 PM](https://github.com/user-attachments/assets/74b52f2a-8d9b-498c-b920-838bee080ee7)

Given this situation, an instance with locale Chinese, `#locale=ko` (Korean)

There's a logic in `i18n.js` that will change the ttag locale to instance locale then back to users locale. This messes up the locale set by calling `loadLocalization` (Korean) since `window.MetabaseUserLocalization` would still be the same value as the instance locale (Chinese) since there's no user in static embed or public link.

It makes sense to set `window.MetabaseUserLocalization` to the downloaded locale in this case, since this represents the user locale set by the hash option.